addons/html_editor/static/src/others/embedded_components/backend/syntax_highlighting/syntax_highlighting.js

```diff
@@ -57,176 +80,57 @@ export class EmbeddedSyntaxHighlightingComponent extends Component {
             this.textarea = this.textareaRef.el;
             this.document = this.textarea.ownerDocument;
 
-            // Load the CSS.
-            const theme = cookie.get("color_scheme") === "dark" ? "okaida" : "default";
-            const prismStyleLink = document.createElement("link");
-            prismStyleLink.rel = "stylesheet";
-            prismStyleLink.href = `/web/static/lib/prismjs/themes/${theme}.css`;
-            this.document.head.append(prismStyleLink);
-
             // Activate and focus the textarea if required.
             if (this.props.autofocus) {
-                this.state.isActive = true;
                 if (this.textarea !== this.document.activeElement) {
                     this.textarea.focus();
                     this.props.onTextareaFocus();
                 }
             }
-
-            // Set the initial values and highlight the pre.
-            this.initialValue =
-                this.state.host.dataset.syntaxHighlightingValue || this.props.initialValue;
-            this.initialLanguageId = this.state.host.dataset.languageId || DEFAULT_LANGUAGE_ID;
-            this.commitToHost(
-                {
-                    value: this.initialValue,
-                    languageId: this.initialLanguageId,
-                },
-                false
-            ).then((didHighlight) => !didHighlight && this.highlight());
-
-            // Ensure the values of the dataset and the content match.
-            this.observer = new MutationObserver((mutations) => {
-                if (mutations.some((mutation) => mutation.oldValue !== null)) {
-                    // Prevent UNDO from returning to a state where the dataset
-                    // was not yet defined.
-                    if (!("syntaxHighlightingValue" in this.state.host.dataset)) {
-                        this.state.host.dataset.syntaxHighlightingValue = this.initialValue;
-                    }
-                    if (!("languageId" in this.state.host.dataset)) {
-                        this.state.host.dataset.languageId = this.initialLanguageId;
-                    }
-                    this.highlight();
-                }
-            });
-            this.observer.observe(this.state.host, {
-                attributeFilter: ["data-syntax-highlighting-value", "data-language-id"],
-                attributeOldValue: true,
-            });
-        });
-        onWillDestroy(() => {
-            this.observer?.disconnect();
+            didAssetsLoad.then(this.highlight.bind(this));
```

Comment:
you don't have to `then` if `loadPrism` promise is returned during `onWillStart`